### ISSUE TITLE
Episodes deleted in Plex not deleting in Kodi

### ISSUE THREAD

**Issue by jt196** [croneter/PlexKodiConnect#1239]
So, maybe I'm being dumb but I can't figure out how to change this behaviour.

I've set Plex up to auto delete watched episodes. Only when they're deleted in Plex, that is not updated in Kodi. 

If I try running the Clean Library command within Kodi, it's an absolute nightmare to get things up and running again. The entire library disappears from Kodi, and I've got to do a bunch of restarts, logging in and out etc to get everything back in. 

Shouldn't PKC just show whatever is in the Plex library?  I've set up direct links as well as I thought this might be something to do with what was interfering with the clean library process.

Happy to provide logs if this isn't expected behaviour.

**Comment by croneter** [comment]
Yes, PlexKodiConnect should automatically delete! You probably screwed something up with the Kodi clean library command.

In the PlexKodiConnect settings under Advanced, choose Reset the Kodi database (in red). Then YES, then NO. That should fix the issue.

**Comment by jt196** [comment]
I just ran the usual clean library command from the menu - not sure how I could have screwed it up. This has happened several times now, I run it and everything is deleted from the system. The episodes go, but the TV shows don't. 

Would this have a detrimental effect if they're not directly linked (rather Add-on)? I'm pretty sure the first few times they were? 

So I've got a couple of weird behaviours: 

1. Episodes aren't being deleted when Plex deletes them
2. Clean library removes episodes and films. 

I did try the clean library command as well but it didn't work. I'll give it a go again with the yes no bit (not sure what the prompts are saying). This is all running on a Shield TV so getting the logs out is a bit fiddly. If I still have issues I'll find them and post here.

**Comment by jt196** [comment]
@croneter so did another check today. Deleted a film, film deleted in library, deleted TV episode - doesn't delete in library. Both of these I did a Plex rescan and they were removed from the Plex library. 

Here's the parts of the log that reference the deleted TV episode. I've changed the file name to <tv_episode_here>. 

There's some reference to `auto delete: false` it's the only thing I can see that I can understand.  Think it's all related to me trying to play back the episode. 

```
2020-11-17 17:06:52.376 T:24640   DEBUG: StartScript - calling plugin PlexKodiConnect Helper TV Shows('plugin://plugin.video.plexkodiconnect.tvshows/212593/','8','?mode=play&plex_id=212595&plex_type=episode&filename=<tv_episode_here>','resume:false')
2020-11-17 17:06:52.376 T:24982   DEBUG: Thread LanguageInvoker start, auto delete: false
2020-11-17 17:06:52.376 T:24982    INFO: initializing python engine.
2020-11-17 17:06:52.376 T:24982   DEBUG: CPythonInvoker(21, /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows/default.py): start processing
2020-11-17 17:06:52.397 T:24983   DEBUG: Thread scriptobs start, auto delete: false
2020-11-17 17:06:52.398 T:24982   DEBUG: -->Python Interpreter Initialized<--
2020-11-17 17:06:52.398 T:24982   DEBUG: CPythonInvoker(21, /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows/default.py): the source file to load is "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows/default.py"
2020-11-17 17:06:52.398 T:24982   DEBUG: CPythonInvoker(21, /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows/default.py): setting the Python path to /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows:/:/data/app/org.xbmc.kodi--i9IOO3Hr5vOvxbPlV918Q==/base.apk/assets/python2.7/lib/python27.zip:/data/app/org.xbmc.kodi--i9IOO3Hr5vOvxbPlV918Q==/base.apk/assets/python2.7/lib/python2.7:/data/app/org.xbmc.kodi--i9IOO3Hr5vOvxbPlV918Q==/base.apk/assets/python2.7/lib/python2.7/site-packages:/data/app/org.xbmc.kodi--i9IOO3Hr5vOvxbPlV918Q==/base.apk/assets/python2.7/lib/python2.7/plat-linux2:/data/app/org.xbmc.kodi--i9IOO3Hr5vOvxbPlV918Q==/base.apk/assets/python2.7/lib/python2.7/lib-tk:/data/app/org.xbmc.kodi--i9IOO3Hr5vOvxbPlV918Q==/base.apk/assets/python2.7/lib/python2.7/lib-old:/data/app/org.xbmc.kodi--i9IOO3Hr5vOvxbPlV918Q==/base.apk/assets/python2.7/lib/python2.7/lib-dynload
2020-11-17 17:06:52.398 T:24982   DEBUG: CPythonInvoker(21, /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows/default.py): entering source directory /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows
2020-11-17 17:06:52.398 T:24982   DEBUG: CPythonInvoker(21, /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows/default.py): instantiating addon using automatically obtained id of "plugin.video.plexkodiconnect.tvshows" dependent on version 2.1.0 of the xbmc.python api
2020-11-17 17:06:52.427 T:24982  NOTICE: PLEX.TVSHOWS: PKC add-on for tv shows started
2020-11-17 17:06:52.428 T:24982   DEBUG: PLEX.TVSHOWS: Full sys.sys.argv received: ['plugin://plugin.video.plexkodiconnect.tvshows/212593/', '8', '?mode=play&plex_id=212595&plex_type=episode&filename=<tv_episode_here>', 'resume:false']
2020-11-17 17:06:52.428 T:24982   DEBUG: PLEX.transfer: Waiting for transfer from main
2020-11-17 17:06:52.522 T:24747   DEBUG: PLEX.playback_starter: Starting PlaybackTask with ?mode=play&plex_id=212595&plex_type=episode&filename=<tv_episode_here>&handle=8&resume=0
2020-11-17 17:06:52.523 T:24747   DEBUG: PLEX.playback_starter: Received mode: play, params: {u'handle': u'8', u'resume': u'0', u'filename': u'His Dark Materials S02E02 1080p HDTV DD5.1 x264-OM2[eztv.re].mkv', u'mode': u'play', u'plex_type': u'episode', u'plex_id': u'212595'}
2020-11-17 17:06:52.523 T:24747   DEBUG: PLEX.playback: playback_triage called with plex_id 212595, plex_type episode, path None, resolve True, resume False
2020-11-17 17:06:52.597 T:24640   DEBUG: ------ Window Init (DialogBusy.xml) ------
2020-11-17 17:06:52.598 T:24747   DEBUG: PLEX.playback: Could not apply playlist hack! Probably Widget playback
2020-11-17 17:06:52.599 T:24747   DEBUG: PLEX.playback: playQueue position 0 for {'playlistid': 1, 'id': None, 'version': None, 'type': 'video', 'selectedItemID': None, 'selectedItemOffset': None, 'shuffled': 0, 'repeat': 0, 'kodi_playlist_playback': False, 'pkc_edit': False, 'items': []}
2020-11-17 17:06:52.599 T:24747   DEBUG: PLEX.playback: PKC playqueue yet empty, need to initialize playback
2020-11-17 17:06:52.599 T:24747   DEBUG: PLEX.playback: Initializing PKC playback
2020-11-17 17:06:52.665 T:24747 WARNING: PLEX.download: Unknown answer from PMS https://192-168-1-77.3198cf502d5447af959b9e99c718fa60.plex.direct:32400/library/metadata/212595?includeRelated=0&includeExtras=1&includeReviews=1&skipRefresh=1&checkFiles=0 with status code 404: <html><head><title>Not Found</title></head><body><h1>404 Not Found</h1></body></html>
2020-11-17 17:06:52.665 T:24747   ERROR: PLEX.plex_functions: Error retrieving metadata for {server}/library/metadata/212595
2020-11-17 17:06:52.666 T:24747   ERROR: PLEX.playback: Could not get a PMS xml for plex id 212595
2020-11-17 17:06:52.666 T:24747   DEBUG: PLEX.transfer: Sending: True
2020-11-17 17:06:52.666 T:24747   DEBUG: PLEX.transfer: Waiting for transfer from default
2020-11-17 17:06:52.734 T:24982   DEBUG: PLEX.transfer: Received
2020-11-17 17:06:52.735 T:24982   DEBUG: PLEX.transfer: Sending: True
2020-11-17 17:06:52.735 T:24982  NOTICE: PLEX.TVSHOWS: PKC add-on for tv shows stopped
```
Here's what happens when I try to play the episode that is missing from the disk:
```
2020-11-17 17:06:52.522 T:24747   DEBUG: PLEX.playback_starter: Starting PlaybackTask with ?mode=play&plex_id=212595&plex_type=episode&filename=<TV_episode_here>&handle=8&resume=0
2020-11-17 17:06:52.523 T:24747   DEBUG: PLEX.playback_starter: Received mode: play, params: {u'handle': u'8', u'resume': u'0', u'filename': u'<TV_episode_here>', u'mode': u'play', u'plex_type': u'episode', u'plex_id': u'212595'}
2020-11-17 17:06:52.523 T:24747   DEBUG: PLEX.playback: playback_triage called with plex_id 212595, plex_type episode, path None, resolve True, resume False
2020-11-17 17:06:52.597 T:24640   DEBUG: ------ Window Init (DialogBusy.xml) ------
2020-11-17 17:06:52.598 T:24747   DEBUG: PLEX.playback: Could not apply playlist hack! Probably Widget playback
2020-11-17 17:06:52.599 T:24747   DEBUG: PLEX.playback: playQueue position 0 for {'playlistid': 1, 'id': None, 'version': None, 'type': 'video', 'selectedItemID': None, 'selectedItemOffset': None, 'shuffled': 0, 'repeat': 0, 'kodi_playlist_playback': False, 'pkc_edit': False, 'items': []}
2020-11-17 17:06:52.599 T:24747   DEBUG: PLEX.playback: PKC playqueue yet empty, need to initialize playback
2020-11-17 17:06:52.599 T:24747   DEBUG: PLEX.playback: Initializing PKC playback
2020-11-17 17:06:52.665 T:24747 WARNING: PLEX.download: Unknown answer from PMS https://192-168-1-77.3198cf502d5447af959b9e99c718fa60.plex.direct:32400/library/metadata/212595?includeRelated=0&includeExtras=1&includeReviews=1&skipRefresh=1&checkFiles=0 with status code 404: <html><head><title>Not Found</title></head><body><h1>404 Not Found</h1></body></html>
2020-11-17 17:06:52.665 T:24747   ERROR: PLEX.plex_functions: Error retrieving metadata for {server}/library/metadata/212595
2020-11-17 17:06:52.666 T:24747   ERROR: PLEX.playback: Could not get a PMS xml for plex id 212595
2020-11-17 17:06:52.666 T:24747   DEBUG: PLEX.transfer: Sending: True
2020-11-17 17:06:52.666 T:24747   DEBUG: PLEX.transfer: Waiting for transfer from default
2020-11-17 17:06:52.734 T:24982   DEBUG: PLEX.transfer: Received
2020-11-17 17:06:52.735 T:24982   DEBUG: PLEX.transfer: Sending: True
2020-11-17 17:06:52.735 T:24982  NOTICE: PLEX.TVSHOWS: PKC add-on for tv shows stopped
2020-11-17 17:06:52.735 T:24982    INFO: CPythonInvoker(21, /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows/default.py): script successfully run
2020-11-17 17:06:52.736 T:24982   DEBUG: onExecutionDone(21, /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.plexkodiconnect.tvshows/default.py)
2020-11-17 17:06:52.744 T:24640   DEBUG: ------ Window Deinit (DialogBusy.xml) ------
2020-11-17 17:06:52.745 T:24983   DEBUG: Thread scriptobs 24983 terminating
2020-11-17 17:06:52.881 T:24982    INFO: Python interpreter stopped
2020-11-17 17:06:52.881 T:24982   DEBUG: Thread LanguageInvoker 24982 terminating
2020-11-17 17:06:52.881 T:24747   DEBUG: PLEX.transfer: Received
2020-11-17 17:06:52.882 T:24747   DEBUG: PLEX.playback_starter: Finished PlaybackTask
2020-11-17 17:06:52.891 T:24640   DEBUG: ------ Window Init (DialogNotification.xml) ------
2020-11-17 17:06:54.017 T:24666   DEBUG: GetImageHash - unable to stat url http://www.primativvu.it/wordpress/wp-content/uploads/2015/03/Prima-Tivvu.png
2020-11-17 17:06:54.018 T:24666    INFO: easy_acquire - Created session to http://www.nododigordio.org
2020-11-17 17:06:55.100 T:24666   DEBUG: GetImageHash - unable to stat url http://www.nododigordio.org/assets/uploads/2018/10/RL_06-300x168.png
2020-11-17 17:06:55.103 T:24666    INFO: easy_acquire - Created session to https://www.radiosienatv.it
2020-11-17 17:06:56.882 T:24640   DEBUG: ------ Window Deinit (DialogNotification.xml) ------
2020-11-17 17:06:57.047 T:24771   DEBUG: PLEX.plexgdm: Recieved UDP packet from [('192.168.1.77', 36836)] containing [M-SEARCH * HTTP/1.1]
2020-11-17 17:06:57.047 T:24771   DEBUG: PLEX.plexgdm: Detected client discovery request from ('192.168.1.77', 36836).  Replying
2020-11-17 17:06:57.048 T:24771   DEBUG: PLEX.plexgdm: Sending registration data HTTP/1.0 200 OK
```
I get another warning in the log from the episode, but this is joined by a few other episodes, I think it's the Next Up menu and some issue there. Don't think it's connected:
```
2020-11-17 17:05:55.176 T:24676 WARNING: CreateLoader - unsupported protocol(plugin) in plugin://plugin.video.plexkodiconnect.tvshows/212593/?mode=play&plex_id=212595&plex_type=episode&filename=<tv_episode_here>
```
Pastebin wouldn't let me paste the whole log so here's a temporary [Dropbox link](https://www.dropbox.com/s/5icxcb7basg7vdd/kodi.log?dl=0). It's quite long (4000 lines) so I hope it's of some use!

**Comment by jt196** [comment]
I haven't set up the Direct Paths yet, as it's a bit confusing (Plex is in Docker, video on NAS). If you think that might help with the clean library tasks, I'll give it a go. If you want me to pass some logs with the clean library data, I can do that too. It really shouldn't remove everything from the PKC library, should it?

**Comment by jt196** [comment]
Update: 
- Set up Direct paths, no change. Guessing it's just about how they're playing rather than the DB. 
- Did a little search, thought it might be related to [issue 247](https://github.com/croneter/PlexKodiConnect/issues/247
), so I removed all the widgets from the front menu, but still no joy. Embuary has widgets in the TV section as well, so it could be related to that. 

Guess I'll just do a hard reset once a month...

**Comment by croneter** [comment]
Hopefully fixed in [PKC **beta** version 2.12.5](https://github.com/croneter/PlexKodiConnect/releases/tag/2.12.5). Let me know here if this is not the case

**Comment by jt196** [comment]
OK just loaded the beta version and it looks like it's cleaned up the dead video files! Well done! Will report back/reopen if there's any further issues.
